mantidproject/mantid
# Create file outputs for PDFGui and RMCProfile

**Description of work.**
This PR fixes a bug in the existing `PDFGui` that prevented it from working with histogram data normaly output from polaris.
This PR implements an algorithm `RMCProfile` that saves workspaces with the appropriate header to be read by the RMCProfile package

**Report to:** Helen Playford

**To test:**
The script below will test the two algorithms, Slack me for the config files if I have not already sent them in another PR. 
```
from mantid.simpleapi import *
from isis_powder import polaris, SampleDetails
config_file_path = r"C:/Users/wey38795/Documents/polaris-calculate-pdf/polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="Test", mode="PDF")  

sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si')
polaris.set_sample_details(sample=sample_details)
polaris.create_vanadium(first_cycle_run_no="98532",
                        multiple_scattering=False)
polaris.focus(run_number="98533", input_mode='Summed')

q_lim = [[2.5, 3, 4, 6, 7], [3.5, 5, 7, 11, 40]]
polaris.create_total_scattering_pdf(run_number='98533', 
                                    merge_banks=True, 
                                    q_lims=q_lim, 
                                    output_binning=[0,0.1,20],
                                    pdf_type='G(r)', 
                                    freq_params=[1])

SaveRMCProfile(InputWorkspace="merged_ws", InputType='S(Q)-1', Title='98533', Filename='C:/polaris_output/rmc.fq')
SavePDFGui(InputWorkspace="pdf_output", Filename="C:/polaris_output/pdf.gr")
```

There should be two files output. 
pdf.gr should have the header:
```
#Comment: neutron, Qmin=2.51279, Qmax=39.9894
##### start data
#S 1 - PDF from Mantid 4.2.20200207.1050
#L r G(r) dr dG(r)
```
rmc.fq should have the the header:
```
5170
rmc S(Q)-1 #  98533
```

Fixes #27774

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
